js/ui/grid_core/ui.grid_core.columns_controller.js

```diff
@@ -1586,24 +1587,67 @@ module.exports = {
 
                     return expandColumns;
                 },
+                setBandColumnsCache: function() {
+                    var columns = this._columns,
+                        columnChildrenByIndex = {},
+                        columnParentByIndex = {};
+
+                    if(columns) {
+                        columns.forEach(function(column) {
+                            var parentIndex = column.ownerBand,
+                                parent = columns[parentIndex];
+
+                            if(column.colspan) {
+                                column.colspan = undefined;
+                            }
+
+                            if(column.rowspan) {
+                                column.rowspan = undefined;
+                            }
+
+                            if(parent) {
+                                columnParentByIndex[column.index] = parent;
+                            } else {
+                                parentIndex = -1;
+                            }
+
+                            columnChildrenByIndex[parentIndex] = columnChildrenByIndex[parentIndex] || [];
+                            columnChildrenByIndex[parentIndex].push(column);
+                        });
+                    }
+
+                    this._columnChildrenByIndex = columnChildrenByIndex;
+                    this._columnParentByIndex = columnParentByIndex;
+                },
+                getBandColumnsCache: function() {
+                    if(!this._columnChildrenByIndex || !this._columnParentByIndex) {
```

Comment:
Combine _columnChildrenByIndex and _columnParentByIndex variables to _bandColumnsCache